Assets/Scripts/Puzzles/Color/Player.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DN.Puzzle.Color
+{
+	/// <summary>
+	/// This player class is unique to the Color puzzle, and follows
+	/// all the lines that are available in the current node if specified by
+	/// the queue.
+	/// </summary>
+	public class Player : MonoBehaviour
+	{
+		enum State
+		{ 
+			Idle,
+			Navigating,
+			Moving,
+			Waiting,
+			Stuck,
+			Done,
+		}
+
+		[SerializeField] private Node currentNode;
+		private PlayerPathFinding playerPathFinding;
+		private ColorPuzzleSettings colorPuzzleSettings;
+		private ICommandQueue<ColorCommand> commandQueue;
+
+		private bool started;
+		private State currentState = State.Idle; 
+
+		private void Awake()
+		{
+			playerPathFinding = GetComponent<PlayerPathFinding>();
+			colorPuzzleSettings = Resources.Load<ColorPuzzleSettings>("ColorPuzzleSettings");
+		}
+
+		public void StartNavigationQueue(ICommandQueue<ColorCommand> navigationQueue)
+		{
+			commandQueue = navigationQueue;
+			started = true;
+
+			StartCoroutine(RunQueue(OnRunCompleted));
+		}
+
+		private Line Navigate(ColorCommand colorCommand) => playerPathFinding.FindLine(colorCommand, currentNode);
+
+		private void OnRunCompleted()
+		{
+			Debug.Log($"I AM DONE, ENDED WITH STATE: {currentState}");
+		}
+
+		private IEnumerator RunQueue(Action callback)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This method 'RunQueue' has 42 lines, which is greater than the 30 lines authorized. Split it into smaller methods.](https://app.codacy.com/manual/stepperman/pvb-programming/pullRequest?prid=5403921)